src/find_service/vec_publisher_store.rs

```diff
@@ -0,0 +1,160 @@
+use super::proto;
+use super::server::{PublisherNotFoundError, PublisherStore};
+use std::collections::HashMap;
+use std::sync::{Arc, RwLock};
+
+type PublisherStoreMap = Arc<RwLock<HashMap<String, proto::Registration>>>;
+
+#[derive(Clone)]
+pub struct VecPublisherStore {
+    publishers_map: PublisherStoreMap,
+}
+
+impl PublisherStore for VecPublisherStore {
```

Comment:
I remember bringing this up in person; just duplicating this here. Can't this be an impl for `PublisherStoreMap` instead of `VecPublisherStore`. Also, wouldn't this be `HashPublisherStore`?

Reply:
Good call, renamed this to `HasPublisherStore`, also will be testing the  type impl idea now